### PR TITLE
Minor fixes and improvements to GH release workflow

### PR DESCRIPTION
**Pull request by mprimi**
* Make release tag regex stricter (s/`.`/`\.`)
 * Add missing build step
 * Handle RC releases 
 * Update Brew project homepage

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ release:
     name: natscli
   name_template: "Release {{.Version}}"
   draft: true
+  prerelease: auto
```
Comment (mprimi) [inline .goreleaser.yml:9]: This will automatically draft a pre-release when appropriate

File: .goreleaser.yml [hunk 2]
```diff
@@ -61,7 +62,7 @@ brews:
     url_template: "https://github.com/nats-io/natscli/releases/download/{{ .Tag }}/nats-{{.Version}}-{{ .Os }}-{{ .Arch }}{{if .Arm}}{{.Arm}}{{end}}.zip"
     homepage: "https://github.com/nats-io/nats"
     description: "NATS utility"
-    skip_upload: false
+    skip_upload: auto
```
Comment (mprimi) [inline .goreleaser.yml:65]: This will automatically skip publishing when it's a RC

**Comment by mprimi** [comment]
I tested this on my fork, here's examples of what it produces:
https://github.com/mprimi/natscli/releases

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -2,7 +2,8 @@ name: Release
 on:
   push:
     tags:
-      - v[0-9]+.[0-9]+.[0-9]+
+      - v[0-9]+\.[0-9]+\.[0-9]+
+      - v[0-9]+\.[0-9]+\.[0-9]+-RC\.[0-9]+
```
Comment (ripienaar) [inline .github/workflows/release.yaml:6]: lets do same as server for consistency `v2.11.0-preview.2`

File: .goreleaser.yml [hunk 4]
```diff
@@ -61,7 +62,7 @@ brews:
     url_template: "https://github.com/nats-io/natscli/releases/download/{{ .Tag }}/nats-{{.Version}}-{{ .Os }}-{{ .Arch }}{{if .Arm}}{{.Arm}}{{end}}.zip"
     homepage: "https://github.com/nats-io/nats"
```
Comment (ripienaar) [inline .goreleaser.yml:63]: do you mind fixing this url to be `natscli`?

**Review by ripienaar — CHANGES_REQUESTED** [review]

**Review by ripienaar — APPROVED** [review]
LGTM

**Pull request merged**